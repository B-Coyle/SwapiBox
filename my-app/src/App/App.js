import React, {Component} from 'react';
import './App.scss';
import Header from '../Header/Header.js';
import PropTypes from 'prop-types';
import Crawl from '../Crawl/Crawl.js';
import {fetchMovie} from '../ApiCall/apiCalls.js';
import MainContainer from '../MainContainer/MainContainer.js';

class App extends Component{
    constructor(){
        super()
        this.state ={
            allFilms: [],
            randomFilm: {},
            skipCrawl: false 
        }

    }

    componentDidMount() {
       fetchMovie()
       .then(films => this.setState({
           allFilms: films.results
       }, () => this.assignRandomInformation()
       ) )
    }


    assignRandomInformation= () => {
        let randomNumber = Math.floor(Math.random() * this.state.allFilms.length )
        this.setState ({
        randomFilm: this.state.allFilms[randomNumber],
        })
    }

     skipScroll=()=>{
         this.setState({
             skipCrawl: true
         })
    }

    render(){
        if(this.state.skipCrawl === false) {
            return(
                <section className="App mainBody">
                    <header className="headerContainer">
                        <Header />
                    </header>
                    <main className="crawlContainer">
                        <Crawl randomFilm={this.state.randomFilm} />
                        <input className="button skipBtn" type="button" value="skip" onClick={this.skipScroll}/>
                    </main>
                </section>
            )
        } else return(
            <section className="App mainBody">
                <header className="headerContainer">
                    <Header />
                </header>
                <main className="mainCardArea">
                    <MainContainer />
                </main>
            </section>
        )
    }
}

export default App;
