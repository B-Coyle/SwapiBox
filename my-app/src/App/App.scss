*{
  margin: 0;
}
.mainBody {
  color: #fff;
  margin: 0;
  display: grid;
  position: relative;
  grid-template-areas: 
  "header header header"
  "main main main"
}
.buttonContainer{
  grid-area: nav;
}

.crawlContainer{
  grid-area: main;
}

.mainCardArea{
  grid-area: main;
}