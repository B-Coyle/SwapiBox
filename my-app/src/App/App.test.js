import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import App from './App';


describe('App', () => {
// beforeEach(){
//   const wrapper = shallow(<App />)
// }
//double check wrapper setup is correct. Need to make this global so I don't have to reference in each 

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches the snapshot', () => {
  const wrapper= shallow(<App />)
  expect(wrapper).toMatchSnapshot();
});

it('should have an initial state that is an empty array, an empty object, and is false', () => {
  const wrapper = shallow(<App />)
  expect(wrapper.instance().state).toEqual({
    allFilms: [],
    randomFilm: {},
    skipCrawl: false })
});
//set this one above to a variable and set expected to equal that variable

it.skip('should invoke componentDidMount', () => {
  const wrapper= shallow(<App />)
  wrapper.setState({
    allFilms: [{title: 'SW'}]
  })
  expect(wrapper.state('allFilms')).toEqual([]);
  wrapper.instance().componentDidMount();
  expect(wrapper.state('allFilms')).toEqual([{'title': 'SW'}]);
});

it('should update the skipCrawl state when skipCrawl is called', () => {
  const wrapper = shallow(<App />)
  expect(wrapper.state('skipCrawl')).toEqual(false);
  wrapper.instance().skipScroll();
  expect(wrapper.state('skipCrawl')).toEqual(true);
});

it('should update the randomFilm state when assignRandomInformation is called', () => {
  const wrapper = shallow(<App />)
  // const mockFilm = [randomNumber]
  wrapper.setState({
    allFilms: [{title: 'SW'}, {title: 'SW'},{title: 'SW'},{title: 'SW'},{title: 'SW'},{title: 'SW'},{title: 'SW'},]
  })
  expect(wrapper.state('randomFilm')).toEqual({});
  wrapper.instance().assignRandomInformation();
  expect(wrapper.state('randomFilm')).toEqual({title: 'SW'});
 });
})