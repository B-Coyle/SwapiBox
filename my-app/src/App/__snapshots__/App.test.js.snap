// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<article
  className="App mainBody"
>
  <div
    className="headerContainer"
  >
    <Header />
  </div>
  <div
    className="crawlContainer"
  >
    <MyCrawlComponent
      randomFilm={Object {}}
    />
    <input
      className="button skipBtn"
      onClick={[Function]}
      type="button"
      value="skip"
    />
  </div>
</article>
`;
